week06/employee_test.go

```diff
@@ -0,0 +1,111 @@
+package week06
+
+import (
+	"testing"
+)
+
+func Test_Employee_InValid(t *testing.T) {
+	t.Parallel()
+
+	tcs := []struct {
+		description string
+		employee    Employee
+		expected    error
+	}{
+		{
+			description: "Zero_Employee_Number",
+			employee:    Employee(0),
+			expected:    ErrInvalidEmployee(0),
+		},
+		{
+			description: "Negative_Employee_Number",
+			employee:    Employee(-9),
+			expected:    ErrInvalidEmployee(-9),
+		},
+		{
+			description: "Positive_Employee_Number",
+			employee:    Employee(3),
+			expected:    nil,
+		},
+	}
+
+	for _, tc := range tcs {
+
+		t.Run(tc.description, func(t *testing.T) {
+
+			err := tc.employee.IsValid()
+
+			if err != nil {
+				if tc.expected.Error() != err.Error() {
+					t.Fatalf("expected : %#v, got : %#v", tc.expected.Error(), err.Error())
+				}
+			}
+		})
+	}
+}
+
+func Test_Employee_Work(t *testing.T) {
+	t.Parallel()
+
+	m := NewManager()
```

Comment:
You to make sure to clean your resources. Currently, when ever you mark a test as failed it is leaking resources because you are not calling `m.Stop` to free them up. Make sure to defer a call to `m.Stop` immediately to make sure that regardless of how your tests exit, you clean up your resources.

```suggestion
	m := NewManager()
	defer m.Stop()
```